vmm/src/memory_manager.rs

```diff
@@ -1566,6 +1546,18 @@ impl MemoryManager {
     // pages touched during our bulk copy are tracked.
     pub fn start_memory_dirty_log(&self) -> std::result::Result<(), MigratableError> {
         for r in &self.guest_ram_mappings {
+            let user_addr = self
+                .guest_memory()
+                .memory()
+                .get_host_address(GuestAddress(r.gpa))
+                .unwrap();
+            self.vm
+                .start_dirty_log(r.slot, r.gpa, r.size, user_addr as u64)
+                .map_err(|e| {
+                    MigratableError::MigrateSend(anyhow!("Error starting VM dirty log {}", e))
+                })?;
+
+            // Clear the dirty pages bitmap
```

Comment:
Is it still necessary to clear the dirty log bits if we're only enabling on demand?

Reply:
Also, maybe clearing the bits could be it's own API entry point (is that easier for MSHV too.)?

Reply:
I agree we shouldn't need to clear anything since we just enabled the logging right before. Clearing only made sense for the previous behavior where the logging had been enabled since the VM was started.

Reply:
Right. But we don't want to control that from vmm/ any other module. So we always clear that in the hypervisor crate.

Reply:
Agree. No need to clear the bitmap log right after start the dirty log. PR updated.

Reply:
Also, I think `mshv` may need an extra API to "set bits" (using flag `DIRTY_BITMAP_SET_DIRTY`), with what `mshv` crate provides now. 

Based on what I learnt from @russell-islam , the `mshv` crate only allows you to enable/disable the dirty pages logging for all guest memory regions at once (through its API `enable_dirty_page_tracking`/`disable_dirty_page_tracking` which both take no argument). Meanwhile, when you are done with dirty page logging, the `mshv` crate requires you to "set bits" on each individual memory regions (through its API `get_dirty_log(gpa, size, DIRTY_BITMAP_SET_DIRTY)`).  

If the "set bits" API works on the same granularity of `disable_dirty_page_tracking` (that applies to all guest memory regions at once), we will not need an extra API and can do both as a part of the existing API `stop_dirty_log()`. @russell-islam Is this something that we can add to the `mshv` crate? If it is feasible, it will also solve my concern on we should "set bits" for all guest memory regions, instead only for guest RAM regions (which is the behavior of your PR #2887).